tensorflow/python/ops/array_ops.py

```diff
@@ -2515,13 +2539,24 @@ def where(condition, x=None, y=None, name=None):
   has the same shape as `x` and `y`, then it chooses which element to copy from
   `x` and `y`.
 
+  If `broadcast` is True, then values of `x`, `y` and `condition` are
```

Comment:
It sounds like the new behavior is backwards compatible?  Why hide it behind a flag if it doesn't break existing usage?

Reply:
Thanks @ebrevdo. Initially I though the behaviors were different with respect to `broadcast`. Now think again it might be possible to not break the old behavior, while at the same time extend the `broadcast` 👍 . Let me take a look and update the PR.